libcudacxx/include/cuda/__memory_resource/resource_ref.h

```diff
@@ -28,14 +28,49 @@
 #  include <cuda/std/__concepts/_One_of.h>
 #  include <cuda/std/__concepts/all_of.h>
 #  include <cuda/std/__memory/addressof.h>
+#  include <cuda/std/__type_traits/integral_constant.h>
 #  include <cuda/std/__type_traits/is_base_of.h>
+#  include <cuda/std/__type_traits/is_nothrow_move_constructible.h>
+#  include <cuda/std/__utility/move.h>
 #  include <cuda/std/cstddef>
 #  include <cuda/stream_ref>
 
 #  if _CCCL_STD_VER >= 2014
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA_MR
 
+union _AnyResourceStorage
+{
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr _AnyResourceStorage(void* __ptr = nullptr) noexcept
+      : __ptr_(__ptr)
+  {}
+
+  void* __ptr_;
+  char __buf_[3 * sizeof(void*)];
```

Comment:
Remark: I believe all of the current resources we provide fit that box, but we might need reach out to other folks, what a good size is.

For now this should be good